prometheus-metrics-model/src/main/java/io/prometheus/metrics/model/snapshots/DataPointSnapshot.java

```diff
@@ -1,19 +1,30 @@
 package io.prometheus.metrics.model.snapshots;
 
 public abstract class DataPointSnapshot {
+  private final String metricName;
```

Comment:
let's add https://mvnrepository.com/artifact/com.google.code.findbugs/jsr305/3.0.2 (complile only) and annotate with `@Nullable`